tt_metal/impl/dispatch/kernels/vc_eth_tunneler.cpp

```diff
@@ -0,0 +1,298 @@
+        (get_compile_time_arg_val(42) & 0xFF),
+        (get_compile_time_arg_val(43) & 0xFF)
+    };
+
+constexpr uint32_t remote_sender_y[MAX_TUNNEL_LANES] =
+    {
+        (get_compile_time_arg_val(34) >> 8) & 0xFF,
+        (get_compile_time_arg_val(35) >> 8) & 0xFF,
+        (get_compile_time_arg_val(36) >> 8) & 0xFF,
+        (get_compile_time_arg_val(37) >> 8) & 0xFF,
+        (get_compile_time_arg_val(38) >> 8) & 0xFF,
+        (get_compile_time_arg_val(39) >> 8) & 0xFF,
+        (get_compile_time_arg_val(40) >> 8) & 0xFF,
+        (get_compile_time_arg_val(41) >> 8) & 0xFF,
+        (get_compile_time_arg_val(42) >> 8) & 0xFF,
+        (get_compile_time_arg_val(43) >> 8) & 0xFF
+    };
+
+constexpr uint32_t remote_sender_queue_id[MAX_TUNNEL_LANES] =
+    {
+        (get_compile_time_arg_val(34) >> 16) & 0xFF,
+        (get_compile_time_arg_val(35) >> 16) & 0xFF,
+        (get_compile_time_arg_val(36) >> 16) & 0xFF,
+        (get_compile_time_arg_val(37) >> 16) & 0xFF,
+        (get_compile_time_arg_val(38) >> 16) & 0xFF,
+        (get_compile_time_arg_val(39) >> 16) & 0xFF,
+        (get_compile_time_arg_val(40) >> 16) & 0xFF,
+        (get_compile_time_arg_val(41) >> 16) & 0xFF,
+        (get_compile_time_arg_val(42) >> 16) & 0xFF,
+        (get_compile_time_arg_val(43) >> 16) & 0xFF
+    };
+
+constexpr DispatchRemoteNetworkType remote_sender_network_type[MAX_TUNNEL_LANES] =
+    {
+        static_cast<DispatchRemoteNetworkType>((get_compile_time_arg_val(34) >> 24) & 0xFF),
+        static_cast<DispatchRemoteNetworkType>((get_compile_time_arg_val(35) >> 24) & 0xFF),
+        static_cast<DispatchRemoteNetworkType>((get_compile_time_arg_val(36) >> 24) & 0xFF),
+        static_cast<DispatchRemoteNetworkType>((get_compile_time_arg_val(37) >> 24) & 0xFF),
+        static_cast<DispatchRemoteNetworkType>((get_compile_time_arg_val(38) >> 24) & 0xFF),
+        static_cast<DispatchRemoteNetworkType>((get_compile_time_arg_val(39) >> 24) & 0xFF),
+        static_cast<DispatchRemoteNetworkType>((get_compile_time_arg_val(40) >> 24) & 0xFF),
+        static_cast<DispatchRemoteNetworkType>((get_compile_time_arg_val(41) >> 24) & 0xFF),
+        static_cast<DispatchRemoteNetworkType>((get_compile_time_arg_val(42) >> 24) & 0xFF),
+        static_cast<DispatchRemoteNetworkType>((get_compile_time_arg_val(43) >> 24) & 0xFF)
+    };
+
+
+constexpr uint32_t test_results_buf_addr_arg = get_compile_time_arg_val(44);
+constexpr uint32_t test_results_buf_size_bytes = get_compile_time_arg_val(45);
+
+// careful, may be null
+tt_l1_ptr uint32_t* const test_results = reinterpret_cast<tt_l1_ptr uint32_t*>(test_results_buf_addr_arg);
+
+constexpr uint32_t timeout_cycles = get_compile_time_arg_val(46);
+constexpr uint32_t inner_stop_mux_d_bypass = get_compile_time_arg_val(47);
+
+void kernel_main() {
+    rtos_context_switch_ptr = (void (*)())RtosTable[0];
+    //noc_init();
+
+    write_test_results(test_results, PQ_TEST_STATUS_INDEX, PACKET_QUEUE_TEST_STARTED);
```

Comment:
Not super familiar with mux/demux kernels and I know this api exists rn but is `write_test_results` a good name for what this is doing?

Reply:
No, it's not a good name.
Also the kernel is not a test kernel anyway.
This needs some cleaning up.

Reply:
all the packet/tunneling kernels have this common quirk.
I'll clean it up.
write_kernel_status ( ) might be a better name.

Reply:
Fixed